tests/phpunit/tests/rest-api/rest-schema-validation.php

```diff
@@ -396,4 +396,188 @@ public function test_string_max_length() {
 		// three supplementary Unicode code point is to long
 		$this->assertWPError( rest_validate_value_from_schema( $mb_char . $mb_char . $mb_char, $schema ) );
 	}
+
+	public function rest_property_provider() {
+		return array(
+			array(
+				(object) array(
```

Comment:
These don't need to be casted to object. In essentially all cases, the API passes associative arrays. See WP_REST_Request::parse_json_params.